editor/selectors.js

```diff
@@ -918,18 +918,37 @@ export function getSuggestedPostFormat( state ) {
 	const blocks = state.editor.blockOrder;
 
 	let name;
-	// If there is only one block in the content of the post grab its name so
+	// If there is only one block in the content of the post grab its name
 	// so we can derive a suitable post format from it.
 	if ( blocks.length === 1 ) {
 		name = state.editor.blocksByUid[ blocks[ 0 ] ].name;
 	}
 
+	// If there are two blocks in the content and the last one is a text blocks
```

Comment:
What is the intended logic here? Should it really be restricted to posts with two blocks specifically, or is the idea if the block is entirely text except for a first non-text block, it should inherit format based on that first block?

```js
let name;
// If there is only one block in the content of the post grab its name
// so we can derive a suitable post format from it.
if ( uids.length === 1 ) {
	name = getBlock( uids[ 0 ] ).name;
} else if ( uids.length > 1 ) {
	const isMostlyText = uids.slice( 1 ).every( ( uid ) => {
		return getBlock( uid ).name === 'core/paragraph';
	} );

	if ( isMostlyText ) {
		name = getBlock( uids[ 0 ] ).name;
	}
}
```

This is non-trivial logic, so might be worth memoizing, and definitely worth testing.

Reply:
The latter. If there are only two blocks, and the second one is text, the first one determines the format suggestion if it matches one of the post formats.

Example: quote + comment like https://ma.tt/2017/10/gauguin-to-olive-garden/

Reply:
If there are more than two we don't suggest anymore as we consider it a standard post.